ethereumj-core/src/main/java/org/ethereum/listener/BlockReplayer.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) [2016] [ <ether.camp> ]
+ * This file is part of the ethereumJ library.
+ *
+ * The ethereumJ library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ethereumJ library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ethereumJ library. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.ethereum.listener;
+
+import org.apache.commons.collections4.queue.CircularFifoQueue;
+import org.ethereum.core.Block;
+import org.ethereum.core.BlockSummary;
+import org.ethereum.core.TransactionInfo;
+import org.ethereum.core.TransactionReceipt;
+import org.ethereum.db.BlockStore;
+import org.ethereum.db.TransactionStore;
+import org.ethereum.facade.Ethereum;
+import org.ethereum.publish.event.BlockAdded;
+import org.ethereum.sync.BlockDownloader;
+import org.ethereum.util.FastByteComparisons;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigInteger;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.ethereum.publish.event.Events.Type.BLOCK_ADED;
+
+/**
+ * Class capable of replaying stored blocks prior to 'going online' and
+ * notifying on newly imported blocks
+ * <p>
+ * For example of usage, look at {@link org.ethereum.samples.BlockReplaySample}
+ * <p>
+ * Created by Eugene Shevchenko on 07.10.2018.
+ */
+public class BlockReplayer {
```

Comment:
Should we mark its predecessor `BlockReplay` class as a deprecated one?

Reply:
yep, I'll add comment.